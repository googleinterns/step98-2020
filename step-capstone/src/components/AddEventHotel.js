import React, { useEffect } from 'react'
import { useState } from 'react'
import {
    TextField,
    Grid,
    Checkbox,
    FormControlLabel
} from '@material-ui/core';
import {
    MuiPickersUtilsProvider,
    DateTimePicker
} from '@material-ui/pickers';
import DateFnsUtils from "@date-io/date-fns";

export default function AddEventHotel(props) {
    let overwriting = props.data !== undefined;

    const [startDate, handleStartChange] = useState(overwriting ? props.data.startDate : new Date());
    const [endDate, handleEndChange] = useState(overwriting ? props.data.endDate : new Date());
    const [checked, setChecked] = useState(overwriting ? props.data.finalized : false);
    const [title, setTitle] = useState(overwriting ? props.data.title : "");
    const [location, setLocation] = useState(overwriting ? props.data.location : "");
    const [description, setDescription] = useState(overwriting ? props.data.description : "");

    const handleChecked = (e) => {
        setChecked(e.target.checked);
    }

    const handleTitleChange = (e) => {
        setTitle(e.target.value);
    }

    const handleLocationChange = (e) => {
        setLocation(e.target.value);
    }

    const handleDescriptionChange = (e) => {
        setDescription(e.target.value);
    }

    /*
     * Called once change to hook state is complete. Updates data property in AddForm.
     */
    useEffect(() => {
        console.log(title)
        props.onDataChange({
            id: overwriting ? props.data.id : undefined,
            title: title,
            type: props.type,
            startDate: startDate,
            endDate: endDate,
            finalized: checked,
            location: location,
            description: description
        })
    }, [startDate, endDate, checked, title, location, description])

    return (
        <Grid container direction="column">
            <Grid item>
                <TextField
                    id="title"
                    label={"Add Title"}
                    defaultValue={title}
                    fullWidth
                    onChange={handleTitleChange}
                />
            </Grid>
            <Grid item>
                <FormControlLabel
                    control={
                        <Checkbox
                            checked={checked}
                            onChange={handleChecked}
                        />
                    }
                    label="Finalized"
                />
            </Grid>
            <Grid item container direction="row" justify="space-between">
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker
                        label={props.type === "event" ? "Start" : "Check in"}
                        value={startDate}
                        onChange={handleStartChange} />
                </MuiPickersUtilsProvider>
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker
                        label={props.type === "event" ? "End" : "Check out"}
                        value={endDate}
                        onChange={handleEndChange} />
                </MuiPickersUtilsProvider>
            </Grid>
            <Grid item>
                <TextField
                    id="location"
                    label={location.length !== 0 ? location : "Add Location"}
                    fullWidth
                    onChange={handleLocationChange}
                />
            </Grid>
            <Grid item>
                <TextField
                    id="description"
                    label={"Add Description"}
                    defaultValue={description}
                    fullWidth
                    multiline
                    onChange={handleDescriptionChange}
                />
            </Grid>
        </Grid>
    )
}
