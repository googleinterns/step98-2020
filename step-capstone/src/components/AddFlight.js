import React, { useEffect } from 'react'
import { useState } from 'react'
import {
    TextField,
    Grid,
    Checkbox,
    FormControlLabel
} from '@material-ui/core';
import {
    MuiPickersUtilsProvider,
    DateTimePicker
} from '@material-ui/pickers';
import DateFnsUtils from "@date-io/date-fns";

export default function AddFlight(props) {
    let overwriting = props.data !== undefined;

    // Sets values to previous values if editing, otherwise blank slate
    const [departureDate, handleStartChange] = useState(overwriting ? props.data.departureDate : new Date());
    const [arrivalDate, handleEndChange] = useState(overwriting ? props.data.arrivalDate : new Date());
    const [checked, setChecked] = useState(overwriting ? props.data.finalized : false);
    const [departureAirport, setDepartureAirport] = useState(overwriting ? props.data.departureAirport : "");
    const [arrivalAirport, setArrivalAirport] = useState(overwriting ? props.data.arrivalAirport : "")
    const [description, setDescription] = useState(overwriting ? props.data.description : "");

    const handleChecked = (event) => {
        setChecked(event.target.checked);
    }

    const handleDepartureAirport = (e) => {
        setDepartureAirport(e.target.value);
    }

    const handleArrivalAirport = (e) => {
        setArrivalAirport(e.target.value);
    }

    const handleDescriptionChange = (e) => {
        setDescription(e.target.value);
    }

    /*
     * Called once change to hook state is complete. Updates data property in AddForm.
     */
    useEffect(() => {
        props.onDataChange({
            finalized: checked,
            id: overwriting ? props.data.id : undefined,
            type: "flight",
            departureDate: departureDate,
            arrivalDate: arrivalDate,
            departureAirport: departureAirport,
            arrivalAirport: arrivalAirport,
            description: description
        })
        // notifies form if necessary inputs are present
        props.onToggleValidation(!(departureAirport === "" || arrivalAirport === ""))
    }, [departureDate, arrivalDate, checked, departureAirport, arrivalAirport, description])


    // Users must input departure and arrival airports --> renders error input if not
    let renderDepartureAirportField = () => {
        if (departureAirport === "") {
            return (
                <TextField
                    error
                    helperText="Cannot leave field blank"
                    id="departure"
                    label={"Departire IATA code"}
                    defaultValue={departureAirport}
                    onChange={handleDepartureAirport}
                />
            )
        } else {
            return (
                <TextField
                    id="departure"
                    label={"Departire IATA code"}
                    defaultValue={departureAirport}
                    onChange={handleDepartureAirport}
                />
            )
        }
    }

    // Users must input departure and arrival airports --> renders error input if not
    let renderArrivalAirportField = () => {
        if (arrivalAirport === "") {
            return (
                <TextField
                    error
                    helperText="Cannot leave field blank"
                    id="arrival"
                    label={"Arrival IATA code"}
                    defaultValue={arrivalAirport}
                    onChange={handleArrivalAirport}
                />
            )
        } else {
            return (
                <TextField
                    id="arrival"
                    label={"Arrival IATA code"}
                    defaultValue={arrivalAirport}
                    onChange={handleArrivalAirport}
                />
            )
        }
    }

    return (
        <Grid container direction="column">
            <Grid item container direction="row" justify="space-between">
                {renderDepartureAirportField()}
                {renderArrivalAirportField()}
            </Grid>
            <Grid item>

                <FormControlLabel
                    control={
                        <Checkbox
                            checked={checked}
                            onChange={handleChecked}
                        />
                    }
                    label="Finalized"
                />
            </Grid>
            <Grid item container direction="row" justify="space-between">
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker label="Departure" value={departureDate} onChange={handleStartChange} />
                </MuiPickersUtilsProvider>
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker label="Arrival" value={arrivalDate} onChange={handleEndChange} />
                </MuiPickersUtilsProvider>
            </Grid>
            <Grid item>
                <TextField
                    id="description"
                    label={"Add Description"}
                    defaultValue={description}
                    fullWidth
                    multiline
                    onChange={handleDescriptionChange}
                />
            </Grid>
        </Grid>
    )
}

