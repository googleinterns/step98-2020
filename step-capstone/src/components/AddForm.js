import React, { useEffect } from 'react'
import { useState } from 'react'
import {
    Typography,
    Card,
    CardActions,
    CardContent,
    Button,
    TextField,
    Paper,
    Grid,
    Tabs,
    Tab,
    Checkbox
} from '@material-ui/core';
import {
    MuiPickersUtilsProvider,
    DateTimePicker
} from '@material-ui/pickers';
import DateFnsUtils from "@date-io/date-fns";


export default class AddForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: 0,
            onClose: props.onClose,
            onAddItem: props.onAddItem,
            data: {}
        }
        this.handleToggleTab = this.handleToggleTab.bind(this);
        this.handleDataChange = this.handleDataChange.bind(this);
        this.handleCreate = this.handleCreate.bind(this);
    }

    // clears data when tab is changed
    handleToggleTab(event, newVal) {
        this.setState({
            value: newVal,
            data: undefined
        });
    }

    handleDataChange(newData) {
        this.setState({
            data: newData
        })
    }

    getForm() {
        switch (this.state.value) {
            case 0: return <AddEventHotel onDataChange={this.handleDataChange} />
            case 1: return <AddFlight onDataChange={this.handleDataChange} />
            case 2: return <AddEventHotel onDataChange={this.handleDataChange} />
            default: return;
        }
    }

    handleCreate() {
        switch (this.state.value) {
            case 0:
                this.state.onAddItem("event", this.state.data);
                break;
            case 1:
                this.state.onAddItem("flight", this.state.data);
                break;
            case 2:
                this.state.onAddItem("hotel", this.state.data);
                break;
            default: break;
        }
        this.state.onClose();
    }

    render() {
        return (
            <Card id="add-form-container">
                <CardContent>
                    <Paper>
                        <Tabs
                            value={this.state.value}
                            onChange={this.handleToggleTab}
                            variant="fullWidth"
                        >
                            <Tab label="Event" />
                            <Tab label="Flight" />
                            <Tab label="Hotel" />
                        </Tabs>
                    </Paper>
                    {this.getForm()}
                </CardContent>
                <CardActions>
                    <Button onClick={this.state.onClose} size="small">Cancel</Button>
                    <Button onClick={this.handleCreate} size="small">Create</Button>
                </CardActions>
            </Card>
        )
    }
}

function AddEventHotel(props) {
    const [startDate, handleStartChange] = useState(new Date());
    const [endDate, handleEndChange] = useState(new Date());
    const [checked, setChecked] = useState(false);
    const [title, setTitle] = useState("");
    const [location, setLocation] = useState("");
    const [description, setDescription] = useState("");

    const handleChecked = (e) => {
        setChecked(e.target.checked);
    }

    const handleTitleChange = (e) => {
        setTitle(e.target.value);
    }

    const handleLocationChange = (e) => {
        setLocation(e.target.value);
    }

    const handleDescriptionChange = (e) => {
        setDescription(e.target.value);
    }

    /*
     * Called once change to hook state is complete. Updates data property in AddForm.
     */
    useEffect(() => {
        props.onDataChange({
            title: title,
            startDate: startDate,
            endDate: endDate,
            finalized: checked,
            location: location,
            description: description
        })
    }, [startDate, endDate, checked, title, location, description])

    return (
        <Grid container direction="column">
            <Grid item>
                <TextField id="title" label="Add Title" fullWidth onChange={handleTitleChange} />
            </Grid>
            <Grid item>
                <Checkbox
                    checked={checked}
                    onChange={handleChecked}
                />
            </Grid>
            <Grid item container direction="row" justify="space-between">
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker value={startDate} onChange={handleStartChange} />
                </MuiPickersUtilsProvider>
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker value={endDate} onChange={handleEndChange} />
                </MuiPickersUtilsProvider>
            </Grid>
            <Grid item>
                <TextField
                    id="location"
                    label="Add Location"
                    fullWidth onChange={handleLocationChange}
                />
            </Grid>
            <Grid item>
                <TextField
                    id="description"
                    label="Add Description"
                    fullWidth
                    multiline
                    onChange={handleDescriptionChange}
                />
            </Grid>
        </Grid>
    )
}

function AddFlight(props) {
    const [departureDate, handleStartChange] = useState(new Date());
    const [arrivalDate, handleEndChange] = useState(new Date());
    const [checked, setChecked] = useState(false);
    const [departureAirport, setDepartureAirport] = useState("");
    const [arrivalAirport, setArrivalAirport] = useState("")
    const [description, setDescription] = useState("");

    const handleChecked = (event) => {
        setChecked(event.target.checked);
    }

    const handleDepartureAirport = (e) => {
        setDepartureAirport(e.target.value);
    }

    const handleArrivalAirport = (e) => {
        setArrivalAirport(e.target.value);
    }

    const handleDescriptionChange = (e) => {
        setDescription(e.target.value);
    }

    /*
     * Called once change to hook state is complete. Updates data property in AddForm.
     */
    useEffect(() => {
        console.log("changing data");
        props.onDataChange({
            finalized: checked,
            departureDate: departureDate,
            arrivalDate: arrivalDate,
            departureAirport: departureAirport,
            arrivalAirport: arrivalAirport,
            description: description
        })
    }, [departureDate, arrivalDate, checked, departureAirport, arrivalAirport, description])

    return (
        <Grid container direction="column">
            <Grid item container direction="row">
                <TextField
                    id="departure"
                    label="Add departure airport"
                    onChange={handleDepartureAirport}
                />
                <TextField
                    id="arrival"
                    label="Add arrival airport"
                    onChange={handleArrivalAirport}
                />
            </Grid>
            <Grid item>
                <Checkbox
                    checked={checked}
                    onChange={handleChecked}
                />
            </Grid>
            <Grid item container direction="row" justify="space-between">
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker value={departureDate} onChange={handleStartChange} />
                </MuiPickersUtilsProvider>
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <DateTimePicker value={arrivalDate} onChange={handleEndChange} />
                </MuiPickersUtilsProvider>
            </Grid>
            <Grid item>
                <TextField
                    id="description"
                    label="Add Description"
                    fullWidth
                    multiline
                    onChange={handleDescriptionChange}
                />
            </Grid>
        </Grid>
    )
}

