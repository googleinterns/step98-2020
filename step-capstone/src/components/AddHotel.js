import React, { useEffect } from 'react'
import { useState } from 'react'
import {
   TextField,
   Grid,
   Checkbox,
   FormControlLabel,
} from '@material-ui/core';
import {
   MuiPickersUtilsProvider,
   DateTimePicker
} from '@material-ui/pickers';
import DateFnsUtils from "@date-io/date-fns";
 
export default function AddHotel(props) {
   let overwriting = props.data !== undefined;
 
   // Sets values to previous values if editing, otherwise blank slate
    const [startDate, setStartDate] = useState(overwriting ? props.data.startDate : props.startDate);
    const [endDate, setEndDate] = useState(overwriting ? props.data.endDate : props.startDate);
    const [checked, setChecked] = useState(overwriting ? props.data.finalized : false);
    const [title, setTitle] = useState(overwriting ? props.data.title : "");
    const [location, setLocation] = useState(overwriting ? props.data.location : "");
    const [description, setDescription] = useState(overwriting ? props.data.description : "");
 
  
   const handleChecked = (e) => {
       setChecked(e.target.checked);
   }
 
   const handleTitleChange = (e) => {
       setTitle(e.target.value);
   }
 
   const handleLocationChange = (e) => {
       setLocation(e.target.value);
   }
 
   const handleDescriptionChange = (e) => {
       setDescription(e.target.value);
   }
   
   const handleStartDateChange = (newStartDate) => { 
        if (newStartDate >= endDate) {
            var newEndDate = new Date(newStartDate);
            newEndDate.setTime(newStartDate.getTime() + 5*60000);
            setEndDate(newEndDate);
        }
        setStartDate(newStartDate);
    }
    const handleEndDateChange = (newEndDate) => {
        if (startDate >= newEndDate) {
            var newStartDate = new Date(newEndDate);
            newStartDate.setTime(newEndDate.getTime() - 5*60000);
            setStartDate(newStartDate);
        }
        setEndDate(newEndDate);
    }

   /*
    * Called once change to hook state is complete. Updates data property in AddForm.
    */
   useEffect(() => {
 
       props.onDataChange({
           id: overwriting ? props.data.id : undefined,
           title: title,
           type: "hotel",
           startDate: startDate,
           endDate: endDate,
           finalized: checked,
           location: location,
           description: description
       })
 
       //validating input
       if (title === "") {
           props.onToggleValidation(false);
       } else if (checked && location === "") {
           props.onToggleValidation(false);
       } else {
           props.onToggleValidation(true);
       }
   }, [startDate, endDate, checked, title, location, description])
 
   // render error input if no title
   let renderTitleInput = () => {
       if (title === "") {
           return (
               <TextField
                   error
                   helperText="Cannot leave field blank"
                   id="title"
                   label={"Add Title"}
                   defaultValue={title}
                   fullWidth
                   onChange={handleTitleChange}
               />
           )
       } else {
           return (
               <TextField
                   id="title"
                   label={"Add Title"}
                   defaultValue={title}
                   fullWidth
                   onChange={handleTitleChange}
               />
           )
       }
   }
 
   // Render error input if finalized item has no location
   let renderLocationInput = () => {
       if (checked && location === "") {
           return (
               <TextField
                   error
                   helperText="Cannot leave field blank"
                   id="location"
                   label={location.length !== 0 ? location : "Add Location"}
                   fullWidth
                   onChange={handleLocationChange}
               />
           )
       } else {
           return (
               <TextField
                   id="location"
                   label={location.length !== 0 ? location : "Add Location"}
                   fullWidth
                   onChange={handleLocationChange}
               />
           )
       }
   }
 
   return (
       <Grid container direction="column">
           <Grid item>
               {renderTitleInput()}
           </Grid>
           <Grid item>
               <FormControlLabel
                   control={
                       <Checkbox
                           checked={checked}
                           onChange={handleChecked}
                       />
                   }
                   label="Finalized"
               />
           </Grid>
           <Grid item container direction="row" justify="space-between">
               <MuiPickersUtilsProvider utils={DateFnsUtils}>
                   <DateTimePicker
                       label={props.type === "event" ? "Start" : "Check in"}
                       value={startDate}
                       onChange={handleStartDateChange} />
               </MuiPickersUtilsProvider>
               <MuiPickersUtilsProvider utils={DateFnsUtils}>
                   <DateTimePicker
                       label={props.type === "event" ? "End" : "Check out"}
                       value={endDate}
                       onChange={handleEndDateChange} />
               </MuiPickersUtilsProvider>
           </Grid>
           <Grid item>
               {renderLocationInput()}
           </Grid>
           <Grid item>
               <TextField
                   id="description"
                   label={"Add Description"}
                   defaultValue={description}
                   fullWidth
                   multiline
                   onChange={handleDescriptionChange}
               />
           </Grid>
       </Grid>
   )
}
 
 

