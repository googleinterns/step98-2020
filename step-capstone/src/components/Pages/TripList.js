import React from 'react';
import { Grid } from '@material-ui/core';
import TripItemComponent from "../Utilities/TripItemComponent";
import AddTrip from '../TripForms/AddTrip';
import {FirebaseContext} from '../Firebase';
import {
  Redirect
} from "react-router-dom";


class TripList extends React.Component {
  static contextType = FirebaseContext;

  constructor(props) {
    super(props);
    this.state = {
      reference : "/users/" + sessionStorage.getItem("userId") + "/trips/",
      selectedTrip : null,
      trips : [],
    };
    this.handleOpenTrip = this.handleOpenTrip.bind(this);
    this.handleAddTrip = this.handleAddTrip.bind(this);
    this.handleDeleteTrip = this.handleDeleteTrip.bind(this);

  }

  loadTrips() {
    let trips = [];
    this.context.getTripList(this.state.reference).then(tripList => {
      tripList.forEach(trip => {
        trips.push(trip)
      })
      this.setState({ trips: trips })
    })
      .catch(error => {
        console.log("Error Getting Trips")
        console.log(error)
        console.error(error)
      })
  }

  componentDidMount() {
    this.loadTrips();

  }

  handleOpenTrip(tripId) {
    this.setState({ selectedTrip: tripId })
    sessionStorage.setItem("tripId", tripId);
  }

  handleAddTrip(newTrip) {
    this.context.addTrip(this.state.reference, newTrip);
    this.loadTrips();
  }

  handleDeleteTrip(tripId) {
    this.context.deleteTrip(this.state.reference+tripId).then(() => this.loadTrips());
    this.loadTrips();
  }

  render() {
    const trips = this.state.trips;
    return (
      <div>
        {this.state.selectedTrip ? <Redirect to={"/trips/" + this.state.selectedTrip} /> :
          <Grid id="trips"
            container spacing={2}
            style={{
              position: "absolute",
              top: "10%",
              left: "10%"
            }}
          >
            {trips.map((trip) => {
              return (
                <Grid item id="tripItem" key={trip.id}>
                  <TripItemComponent
                    key={trip.id}
                    data={trip.data()}
                    tripId={trip.id}
                    onOpenTrip={this.handleOpenTrip}
                    onDeleteTrip={this.handleDeleteTrip} />
                </Grid>
              );
            })}

            <Grid item>
              <AddTrip
                onAddTrip={this.handleAddTrip} />
            </Grid>

          </Grid>
        }
      </div>
    );
  }
}

export default TripList;