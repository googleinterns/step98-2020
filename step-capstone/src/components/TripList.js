import React from 'react';
import { Grid } from '@material-ui/core'
import TripItemComponent from "./TripItemComponent"
import {FirebaseContext} from './Firebase';
import AddItemButton from './AddItemButton';
import {
  Switch,
  Route,
  Redirect
} from "react-router-dom";


class TripList extends React.Component{
  static contextType = FirebaseContext;

  constructor(props){
    super(props);
    this.state = {
      selectedTrip : null,
      reference : "/users/" + sessionStorage.getItem("userId") + "/trips",
      trips : []};
    this.handleOpenTrip = this.handleOpenTrip.bind(this);
    this.handleAddTrip = this.handleAddTrip.bind(this);
  }

  componentDidMount() {
    let trips = [];
    this.context.getTripList(this.state.reference).then(tripList => {
      tripList.forEach(trip => {
        trips.push(trip)
      })
      this.setState({trips : trips})
    })
    .catch(error => {
      console.log("Error Getting Trips")
      console.error(error)
    });
  }

  handleOpenTrip(tripId) {
    this.setState({selectedTrip : tripId})
    sessionStorage.setItem("tripId",tripId);
  }

  handleAddTrip(trip) {
    this.context.addTrip(this.state.reference, trip)
    .then(() => {
      this.setState({trips : this.state.trips.concat(trip)})
    })
    .catch(error => {
      console.log("Error Adding Item")
      console.error(error)
    });
  }

  render () {
    const trips = this.state.trips;
    return (
      <div>
        {this.state.selectedTrip ? <Redirect to = {"/trips/"+this.state.selectedTrip}/> :
          <div>
            <Grid id="trips" container spacing = {2}> {
              trips.map((trip) => {
                return (
                  <Grid item>
                    <TripItemComponent key={trip.id} data={trip.data()} tripId={trip.id} onOpenTrip={this.handleOpenTrip} />
                  </Grid>
                );
              })
            }
            </Grid>
            <Grid id="add-trip-button">
              <p>Get started on your next advenure</p>
              <AddItemButton onAddItem={this.handleAddTrip} />
            </Grid>
          </div>
        }
      </div>

   );
  }
}

export default TripList;