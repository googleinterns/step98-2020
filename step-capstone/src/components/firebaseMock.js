import React from 'react';
//Firebase Mock Setup 

const londonTestTripData = {
  description: "A trip to London",
  endDate: new Date (2020, 7, 10),
  location: "London",
  startDate: new Date (2020, 7, 7),
  title : "London Trip",
  travelObjects : {
    arrivalAirport: "LON",
    arrivalDate: new Date(2020, 7, 7, 8),
    departureAirport: "MCO",
    departureDate: new Date (2020, 7, 7, 15),
    description: "flight from Orlando to London",
    finalized: false,
    id: 0,
    type: "flight"
  }
}

const londonTestTrip = {
  data : () => {return londonTestTripData}
}

const testTrips = [londonTestTrip]

class firebaseMock {
  getTripList() {
    return new Promise((resolve) => {
      resolve(testTrips);
    });
  }
  getTrip() {
    return new Promise ((resolve) => {
      resolve(londonTestTrip);
    })
  }
}
export default firebaseMock;