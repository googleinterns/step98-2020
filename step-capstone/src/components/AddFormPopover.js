import React from 'react';
import { Typography, Popover, Card, CardActions, CardContent, Button, TextField } from '@material-ui/core';
import AddForm from './AddForm'

export default function AddFormPopver(props) {
    return (
        <div>
            <Popover
                id={props.id}
                open={props.open}
                anchorEl={props.anchorEl}
                onClose={props.onClose}
                anchorOrigin={props.anchorOrigin}
                transformOrigin={props.transformOrigin}
            >
                <AddForm onClose={props.onClose}/>
            </Popover>
        </div>
    )
}