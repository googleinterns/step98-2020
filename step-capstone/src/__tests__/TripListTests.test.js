import React from 'react';
import TripList from '../components/TripList';
import renderer from 'react-test-renderer';
import {FirebaseContext } from '../components/Firebase';
import firebaseMock, {testTrips} from '../components/firebaseMock'


 test('TripList displays correctly', async () => {
  debugger; 
  sessionStorage.setItem("userId", "testUser");
  const tripListComponent = renderer.create(
    <FirebaseContext.Provider value={new firebaseMock()}>
      <TripList/>
    </FirebaseContext.Provider>
  );
  //we need to wait for a second to let promises resolve before taking our snapshot
  await new Promise(res => setTimeout(() => { res() }, 0.5))
  let test = tripListComponent.toJSON();
  
  expect(test).toMatchSnapshot();
});