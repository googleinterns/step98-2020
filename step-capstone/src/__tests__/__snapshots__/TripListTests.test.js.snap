// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripList displays correctly 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    id="trips"
    style={
      Object {
        "left": "10%",
        "position": "absolute",
        "top": "10%",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item"
      id="tripItem"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
            >
              London Trip
            </span>
            <span
              className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              Fri Aug 07 2020 - Mon Aug 10 2020
            </span>
          </div>
        </div>
        <img
          className="MuiCardMedia-root makeStyles-media-2 MuiCardMedia-media MuiCardMedia-img"
          src="https://www.state.gov/wp-content/uploads/2019/04/Japan-2107x1406.jpg"
        />
        <div
          className="MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
          >
            A trip to London
          </p>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Share
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Open
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Delete
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-root-3 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            style={
              Object {
                "fontSize": "100",
                "justify": "center",
                "left": "110px",
                "position": "relative",
                "top": "150px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <div
            aria-hidden={true}
            className="MuiBackdrop-root makeStyles-backdrop-5"
            style={
              Object {
                "opacity": 0,
                "visibility": "hidden",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
